components/mailchimp/sources/common/http-based.js

```diff
@@ -0,0 +1,95 @@
+const base = require("./base");
+
+module.exports = {
+  ...base,
+  props: {
+    ...base.props,
+    // eslint-disable-next-line pipedream/props-description, pipedream/props-label
+    http: {
+      type: "$.interface.http",
+      customResponse: true,
+    },
+    triggeredByUser: {
+      type: "boolean",
+      label: "Trigger on subscriber actions?",
+      description:
+          "If set to true, events will be emitted on subscriber-initiated actions.",
+      default: false,
+    },
+    triggeredByAdmin: {
+      type: "boolean",
+      label: "Trigger on actions by Mailchimp admin?",
+      description:
+          "If set to true, events will be emitted on admin-initiated actions in the web interface.",
+      default: false,
+    },
+    triggeredByApi: {
+      type: "boolean",
+      label: "Trigger via actions on the API?",
+      description:
+          "If set to true, events will be emitted on actions initiated via the API.",
+      default: false,
+    },
+  },
+  hooks: {
+    ...base.hooks,
+    async activate() {
+      const eventsCfg = this.getEventsConfig();
+      const webhook = await this.mailchimp.createWebhook(
```

Comment:
Source [deployment](https://pipedream.com/docs/cli/reference/#pd-dev) fails because an error is thrown when creating a Mailchimp webhook in the `activate` hook.

Reply:
This has been previously addressed.